src/spec.ts

```diff
@@ -33,7 +33,7 @@ export type TopLevel<S extends BaseSpec> = S & TopLevelProperties & {
   config?: Config;
 };
 
-export interface BaseSpec {
+export interface BaseSpecWithoutData {
```

Comment:
BaseSpecWithoutData isn't used any where. 

Maybe better do  the following?
```
export type BaseSpec = Partial<DataMixins> & { 
  ...
}
```

Reply:
Done